### DISCUSSION TITLE
Tool List (Update) - Automático

### DISCUSSION THREAD

**Discussion by bgastaldi** [opencodeco/.github#6] [Ideas]
Bom dia, 

Conforme falei com o @flavioheleno (https://github.com/opencodeco/phpctl/pull/29#issuecomment-1989686732), é possível fazer o download da última versão automaticamente, porém existe a questão da compatibilidade. Eu não enviei o PR porque não sei se esse é o objetivo. Por exemplo, se quiséssemos torná-lo automático, seria mais ou menos assim.

```
...
echo "Installing ComposerRequireChecker"
mkdir -p /usr/local/bin/composer-require-checker && wget --no-check-certificate -qO composer-require-checker.phar $(wget -qO- https://api.github.com/repos/maglnet/ComposerRequireChecker/releases/latest | grep "browser_download_url.*\.phar" | cut -d : -f 2,3 | tr -d '," ') && mv composer-require-checker.phar /usr/local/bin/composer-require-checker && chmod a+x /usr/local/bin/composer-require-checker/composer-require-checker.phar
...
```

No caso como o Flávio citou, poderia ter uma função de update.

Abrs

**Comment by leocavalcante** [comment]
Eu tenho minhas dúvidas, é como ter um `composer.json` com `*`. Entendo que seria mais fácil manter as ferramentas atualizadas, mas não ter essa trava na versão que vai rodar pode gerar alguns problemas com a compatibilidade da versão do PHP, mas principalmente com a imprevisibilidade.

Mas vamos ver o que a comunidade acha :)

**Comment by fhferreira** [comment]
Seria muito bom conseguir "isolar" versões das ferramentas para usar. 
Principalmente em casos onde se tem códigos distintos:
 5.6, 7.0 ... 7.4, 8.0 ... 8.3
